File: Go/0485.go
func findMaxConsecutiveOnes(nums []int) int {
    max := 0
    i := 0
    for i < len(nums) {
        curr := 0
        for i < len(nums) && nums[i] == 1 {
            curr++
            if curr > max {
                max = curr
            }
            i++
        }
        i++
    }

    return max
}

File: C/0485.c
int findMaxConsecutiveOnes(int* nums, int numsSize) {
    int max = 0, i = 0;
    while (i < numsSize) {
        int curr = 0;
        while (i < numsSize && nums[i] == 1) {
            if (++curr > max) max = curr;
            i++;
        }
        i++;
    }
    return max;
}
